IGSIOCalibration/vtkIGSIOPivotCalibrationAlgo.cxx

```diff
@@ -160,45 +150,128 @@ igsioStatus vtkIGSIOPivotCalibrationAlgo::GetPivotPointPosition(double* pivotPoi
 }
 
 //----------------------------------------------------------------------------
-igsioStatus vtkIGSIOPivotCalibrationAlgo::DoPivotCalibration(vtkIGSIOTransformRepository* aTransformRepository/* = NULL*/)
+igsioStatus vtkIGSIOPivotCalibrationAlgo::DoCalibrationInternal(const std::vector<vtkMatrix4x4*>* markerToTransformMatrixArray, double& error)
+{
+  double pivotPoint_Marker[4] = { 0.0, 0.0, 0.0, 0.0 };
+  double pivotPoint_Reference[4] = { 0.0, 0.0, 0.0, 0.0 };
+  vtkNew<vtkMatrix4x4> pivotPointToTool;
+  std::set<unsigned int> outlierIndices;
+  if (this->DoPivotCalibrationInternal(markerToTransformMatrixArray, true, &outlierIndices, pivotPoint_Marker, pivotPoint_Reference, pivotPointToTool) != IGSIO_SUCCESS)
+  {
+    return IGSIO_FAIL;
+  }
+
+  error = this->ComputePivotCalibrationError(markerToTransformMatrixArray, &outlierIndices, pivotPoint_Reference, pivotPointToTool);
+  return IGSIO_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+igsioStatus vtkIGSIOPivotCalibrationAlgo::DoPivotCalibration(vtkIGSIOTransformRepository* aTransformRepository/* = NULL*/, bool autoOrient)
+{
+  if (this->MarkerToReferenceTransformMatrixBuckets.empty())
+  {
+    this->SetErrorCode(CALIBRATION_NOT_ENOUGH_POINTS);
+    return IGSIO_FAIL;
+  }
+
+  if (this->GetMaximumToolOrientationDifferenceDeg() < this->MinimumOrientationDifferenceDeg)
+  {
+    this->SetErrorCode(CALIBRATION_NOT_ENOUGH_VARIATION);
+    return IGSIO_FAIL;
+  }
+
+  if (!this->PivotPointToMarkerTransformMatrix)
+  {
+    vtkNew<vtkMatrix4x4> pivotPointToMarkerTransformMatrix;
+    this->SetPivotPointToMarkerTransformMatrix(pivotPointToMarkerTransformMatrix);
+  }
+
+  double pivotPoint_Marker[4] = { 0.0, 0.0, 0.0, 1.0 };
+  double pivotPoint_Reference[4] = { 0.0, 0.0, 0.0, 1.0 };
+  std::vector<vtkMatrix4x4*> markerToTransformMatrixArray = this->GetAllMarkerToReferenceMatrices();
+  igsioStatus status = this->DoPivotCalibrationInternal(&markerToTransformMatrixArray, autoOrient, &this->OutlierIndices, pivotPoint_Marker, pivotPoint_Reference, this->PivotPointToMarkerTransformMatrix);
+  if (status == IGSIO_SUCCESS)
+  {
+    this->SetErrorCode(CALIBRATION_SUCCESS);
+  }
+  else
+  {
+    this->SetErrorCode(CALIBRATION_FAIL);
+  }
+
+  this->PivotPointPosition_Reference[0] = pivotPoint_Reference[0];
+  this->PivotPointPosition_Reference[1] = pivotPoint_Reference[1];
+  this->PivotPointPosition_Reference[2] = pivotPoint_Reference[2];
+
+  this->ComputePivotCalibrationError();
```

Comment:
it is strange that it cannot affect the returned status

Reply:
Yes, it would be nice if there was a reasonable default threshold in the algorithm and the algorithm would return with failure if the error is above that. (If the algorithm could not decide this then the logic would need to be at the application level, which would not be nice because the check would need to be reimplemented everywhere where the algorithm is used.)